src/main/java/com/arpnetworking/metrics/mad/parsers/GraphitePlaintextToRecordParser.java

```diff
@@ -0,0 +1,262 @@
+    }
+
+    private Record createRecord(
+            final String name,
+            final Number value,
+            final DateTime timestamp,
+            final ImmutableMap<String, String> dimensions) {
+        return new DefaultRecord.Builder()
+                .setId(UUID.randomUUID().toString())
+                .setDimensions(dimensions)
+                .setMetrics(ImmutableMap.of(
+                        name,
+                        new DefaultMetric.Builder()
+                                .setValues(
+                                        Collections.singletonList(
+                                                new Quantity.Builder()
+                                                        .setValue(value.doubleValue())
+                                                        .build()))
+                                .setType(MetricType.GAUGE)
+                                .build()
+                ))
+                .setTime(timestamp)
+                .build();
+    }
+
+    private GraphitePlaintextToRecordParser(final Builder builder) {
+        _globalTags = builder._globalTags;
+        _parseCarbonTags = builder._parseCarbonTags;
+    }
+
+    private final ImmutableMap<String, String> _globalTags;
+    private final boolean _parseCarbonTags;
+
+    private static final Splitter LINE_SPLITTER = Splitter.on('\n').omitEmptyStrings();
+    private static final ThreadLocal<NumberFormat> NUMBER_FORMAT = ThreadLocal.withInitial(NumberFormat::getInstance);
+    private static final Pattern GRAPHITE_PATTERN = Pattern.compile(
+            "^(?<NAME>[^ ]+) (?<VALUE>[^ ]+)( (?<TIMESTAMP>[0-9]+))?$");
+
+    /**
+     * Implementation of <code>Builder</code> for {@link GraphitePlaintextToRecordParser}.
+     */
+    public static final class Builder extends OvalBuilder<GraphitePlaintextToRecordParser> {
+
+        /**
+         * Public constructor.
+         */
+        public Builder() {
+            super(GraphitePlaintextToRecordParser::new);
+        }
+
+        /**
+         * Set global tags. Optional. Cannot be null. Default is an empty map.
+         *
+         * @param value the global tags
+         * @return this {@link Builder} instance
+         */
+        public Builder setGlobalTags(final ImmutableMap<String, String> value) {
+            _globalTags = value;
+            return this;
+        }
+
+        /**
+         * Set parse carbon tags. Optional. Cannot be null. Default is false.
+         *
+         * See: http://graphite.readthedocs.io/en/latest/tags.html
+         *
+         * @param value whether to parse carbon tags
+         * @return this {@link Builder} instance
+         */
+        public Builder setParseCarbonTags(final Boolean value) {
+            _parseCarbonTags = value;
+            return this;
+        }
+
+        @NotNull
+        private ImmutableMap<String, String> _globalTags = ImmutableMap.of();
+        @NotNull
+        private Boolean _parseCarbonTags = false;
```

Comment:
Is there a reason to not do this by default?

Reply:
It's not part of the Graphite spec per say. It's someone's (Carbon's) extension to the schema. While it seems benign (not to mention useful -- way better than the influx extension imo) it is not a standard feature of the protocol and so it's off.

Reply:
The link specified in the comments is actually documentation for Graphite.  Carbon is the name of the daemon that receives the data and writes it to the (whisper) database.  I'm OK with not making it default, but I think we're safe to do so.  It's a super new feature, but is part of core graphite now.